module/icp/algs/modes/gcm.c

```diff
@@ -631,22 +641,26 @@ gcm_init_ctx(gcm_ctx_t *gcm_ctx, char *param,
 
 	if (GCM_IMPL_READ(icp_gcm_impl) != IMPL_CYCLE) {
 		gcm_ctx->gcm_use_avx = GCM_IMPL_USE_AVX;
+		gcm_ctx->gcm_use_avx2 = GCM_IMPL_USE_AVX2;
 	} else {
 		/*
 		 * Handle the "cycle" implementation by creating avx and
 		 * non-avx contexts alternately.
 		 */
 		gcm_ctx->gcm_use_avx = gcm_toggle_avx();
+		gcm_ctx->gcm_use_avx2 = gcm_toggle_avx2();
```

Comment:
Pretty sure the cycle behaviour here isn't correct.

Reply:
Should be fixed in #17061.

Reply:
So does this PR have a dependency on #17061?

Reply:
@tonyhutter I've had another go at the cycle code - since `movbe` only applies to AVX I've modified the toggling for that to not apply to AVX2. Furthermore, if bswap is needed neither accelerated method will be usable.